src/primitives.jl

```diff
@@ -210,8 +210,9 @@ dropnull!(X::NullableVector) = deleteat!(X, find(X.isnull)).values # -> Vector
 
 Returns whether or not any entries of `X` are null.
 """
-anynull(X::Any) = any(_isnull, X)           # -> Bool
-anynull(X::NullableArray) = any(X.isnull)   # -> Bool
+anynull(X::Any) = any(_isnull, X)                                           # -> Bool
+anynull{T}(X::AbstractArray{T}) = T <: Nullable ? any(_isnull, X) : false   # -> Bool
```

Comment:
You should be able to avoid the branch entirely and use dispatch here.
```julia
anynull{T<:Nullable}(X::AbstractArray{T}) = any(_isnull, X)
anynull(X::AbstractArray) = false
```